template/src/lib/sdk/widgets/IsochroneWidget.js

```diff
@@ -0,0 +1,174 @@
+import {
+  Button,
+  Divider,
+  FormControl,
+  Grid,
+  InputLabel,
+  makeStyles,
+  MenuItem,
+  Select,
+  Typography,
+} from '@material-ui/core';
+import { selectOAuthCredentials } from 'config/oauthSlice';
+import { addLayer, removeLayer, setError, setIsolineResult } from 'config/cartoSlice';
+import { launchIsochrone, MODES, RANGES } from 'lib/sdk';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+export default function IsochroneWidget(props) {
+  const dispatch = useDispatch();
+  const oauthCredentials = useSelector(selectOAuthCredentials);
+  const globalCredentials = useSelector((state) => state.carto.credentials);
+  const credentials = oauthCredentials || globalCredentials;
+
+  const [openIsochroneConfig, setOpenIsochroneConfig] = useState(false);
+  const [selectedMode, setSelectedMode] = useState(MODES.CAR);
+  const [selectedRange, setSelectedRange] = useState(RANGES.TEN);
+  const { latLong } = props;
+  const classes = useStyles();
+
+  const handleChangeMode = ({ target }) => {
+    setSelectedMode(target.value);
+  };
+
+  const handleChangeRange = ({ target }) => {
+    setSelectedRange(target.value);
+  };
+
+  const updateIsochrone = useCallback(
+    (isochrone) => {
+      dispatch(setIsolineResult(isochrone));
+    },
+    [dispatch]
+  );
+
+  const clickLaunchHandle = () => {
+    const open = !openIsochroneConfig;
+    setOpenIsochroneConfig(open);
+
+    if (!open) {
+      updateIsochrone(null);
+    }
+  };
+
+  useEffect(() => {
+    dispatch(
+      addLayer({
+        id: 'isolineLayer',
+      })
+    );
+
+    return function cleanup() {
```

Comment:
That's a good addition too, the cleanup for this. I will try to add it to Geocoding too